BUILD

```diff
@@ -2312,6 +2313,13 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "grpc_core_credentials",
+    hdrs = ["include/grpc/credentials.h"],
+    language = "c++",
+    visibility = ["@grpc:core_credentials"],
```

Comment:
One potential problem here is that wrapped languages need access to these APIs.  Please make sure we're not breaking that -- e.g., I think we support building python via bazel.

Reply:
Good point. It looks like the python bazel tests are happy, and one of those builds the auth example https://source.cloud.google.com/results/invocations/9b89d452-140c-459f-9c47-1780350c6fc9/targets. 

`//examples/python/auth:_auth_example_test                                PASSED in 0.9s`